frontend/js/pages/BattleDetail.js

```diff
@@ -21,6 +23,18 @@ function BattleDetail(props) {
     props.getBattle(id);
   }, []);
 
+  if (loading) {
+    return (
+      <img
+        alt="loading"
+        className="img_loading"
+        src="https://media0.giphy.com/media/3oEjI6SIIHBdRxXI40/200.gif"
+      />
+    );
+  }
+  if (error) {
+    return 'Ocurred an error';
+  }
   if (!battle) {
     return '';
   }
```

Comment:
This loading states can be better handled don't you think?

Reply:
Hmm great! I'll improve it!